### PR TITLE
fix(TextEditor): update toolbar correctly on paste

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-2293]
[FX-2449]

### Description

fix the issue when pasting formatted text

### How to test

There were multiple issues with pasting
1. pasting anything threw an error
  - when passing undefined to `quill.getFormat()`, quill internally calls `quill.getSelection(true)`, so it first focuses editor and get latest selection
  
 2. after pasting wrapping `Container` lost focus, even we had a cursor inside the Editor.
 3. quill keeps styling of pasted text by default

- 2. and 3. issue was solved by adding library `quill-smart-paste`, that handles all these issues for us.

4. after pasting there was a styling issue in the editor.
- fixed by adding some styles to `ql-clipboard`. I have copied these styles from the default theme of the quill.

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2293]: https://toptal-core.atlassian.net/browse/FX-2293?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FX-2449]: https://toptal-core.atlassian.net/browse/FX-2449?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso-lab/src/QuillEditor/styles.ts [hunk 1]
```diff
@@ -90,10 +90,21 @@ const editor = {
   }
 }
 
+const clipboard = {
+  '& .ql-clipboard': {
+    left: '-100000px',
```
Comment (denieler) [inline packages/picasso-lab/src/QuillEditor/styles.ts:95]: what is this? is it a common way to solve clipboard issue for quill?

File: packages/picasso-lab/src/QuillEditor/utils/getSelectionChangeHandler/getSelectionChangeHandler.tsx [hunk 2]
```diff
@@ -7,13 +7,10 @@ const getSelectionChangeHandler = (
   onSelectionChange: SelectionHandler
 ) => {
   const handler: SelectionChangeHandler = (range, _, source) => {
+    console.log('selection-change')
```
Comment (denieler) [inline packages/picasso-lab/src/QuillEditor/utils/getSelectionChangeHandler/getSelectionChangeHandler.tsx:10]: ☝️

File: packages/picasso-lab/src/QuillEditor/styles.ts (hunk 1 above)
Comment (TomasSlama) [reply packages/picasso-lab/src/QuillEditor/styles.ts:95]: yes, I copied this from their theme, when pasting some text, it did some weird stuff with styling. This fixes the issue

**Review by OndrejTuma — APPROVED** [review]
LGTM

**Comment by separatio** [comment]
I have one question. `packages/picasso-lab/src/QuillEditor/styles.ts` should we be adding/creating picasso-lab again? I thought it was removed. :thinking: Do we plan on adding it back?

If not, please rebase with master and move stuff around.

File: packages/picasso-lab/src/QuillEditor/hooks/useQuillInstance/useQuillInstance.tsx [hunk 3]
```diff
@@ -15,7 +16,13 @@ export type EditorOptionsType = {
 export const getModules = (): QuillOptionsStatic['modules'] => {
   return {
     clipboard: {
-      matchVisual: false
+      matchVisual: false,
+      allowed: {
+        tags: ['b', 'strong', 'i', 'p', 'br', 'ul', 'ol', 'li'],
```
Comment (LashaJini) [inline packages/picasso-lab/src/QuillEditor/hooks/useQuillInstance/useQuillInstance.tsx:21]: https://user-images.githubusercontent.com/72510037/153384985-3503acff-a8f6-428c-bb59-b98c6f704ee0.mp4

**Review by LashaJini — APPROVED** [review]

File: packages/picasso-lab/src/QuillEditor/hooks/useQuillInstance/useQuillInstance.tsx (hunk 3 above)
Comment (separatio) [reply packages/picasso-lab/src/QuillEditor/hooks/useQuillInstance/useQuillInstance.tsx:21]: Nice catch! :grin:

Comment (LashaJini) [reply packages/picasso-lab/src/QuillEditor/hooks/useQuillInstance/useQuillInstance.tsx:21]:
```suggestion
        tags: ['h3', 'b', 'strong', 'i', 'p', 'br', 'ul', 'ol', 'li'],
```

**Comment by LashaJini** [comment]
> I have one question. `packages/picasso-lab/src/QuillEditor/styles.ts` should we be adding/creating picasso-lab again? I thought it was removed. thinking Do we plan on adding it back?
> 
> If not, please rebase with master and move stuff around.

@separatio picasso-lab still exists on text editor branch. We will refactor it before merging, I guess.

**Comment by separatio** [comment]
I didn't notice we have a feature branch with text editor. My bad.

**Comment by separatio** [comment]
Manually tested. Looks good, but I do have a question.
Should the header/ normal text select be opened when it is clicked, and the text editor was out of focus?
The other items can not be interacted with (such as Bold and Italic) if clicked while the Text Editor is not focused.

Please see the video:
https://user-images.githubusercontent.com/8407461/153386081-901a127b-b416-469e-9709-7985647bf5f7.mp4

**Comment by TomasSlama** [comment]
> Manually tested. Looks good, but I do have a question. Should the header/ normal text select be opened when it is clicked, and the text editor was out of focus? The other items can not be interacted with (such as Bold and Italic) if clicked while the Text Editor is not focused.

@separatio with Daniel we chose the same strategy as Slack is using. So when you are out of editor, toolbar is disabled. When you click inside editor/toolbar, you can use toolbar again, so to me, this is right behaviour

**Pull request merged**